src/resolve/resolved-mdns.c

```diff
@@ -386,7 +386,10 @@ static int on_mdns_packet(sd_event_source *s, int fd, uint32_t revents, void *us
         if (r <= 0)
                 return r;
 
-        if (manager_packet_from_local_address(m, p))
+        /* Refuse traffic from the local host, to avoid query loops. However, allow legacy mDNS
+         * unicast queries through anyway (we never send those ourselves, hence no risk).
+         * i.e. check for the source port nr. */
+        if (p->sender_port == MDNS_PORT && manager_packet_from_local_address(m, p))
```

Comment:
I wonder why this is backported to the stable branch?

Reply:
It fixes a timeout/hang.

Reply:
It introduces an interesting side effect https://github.com/systemd/systemd/pull/32991#discussion_r1685731917 and doesn't fully fix the issue: https://github.com/systemd/systemd/pull/32991

Reply:
@bluca please drop the commit then. It doesn't seem ready for stable.

Reply:
> @bluca please drop the commit then. It doesn't seem ready for stable.

Nah, it implements exactly what Lennart asked, verbatim, and fixes a real issue as per attached bug report - this is just the usual targeted abuse from the usual suspect, it can be safely ignored as always

Reply:
> fixes a real issue as per attached bug report

It does not. https://github.com/systemd/systemd/issues/32990 mentions the other clients like `mquery` and it does not work.

Reply:
I opened https://github.com/systemd/systemd/issues/33806 where the side effect is described. I'll open a separate issue about mDNS queriers not working later.

Reply:
@bluca Please drop the commit, I can reopen the same bug report under my username if that makes it better. There's obviously still major issues with the current behavior so we shouldn't backport this commit just yet.

Reply:
Yeah, the problem raised is valid regardless of how it was brought up. Let's drop this for now. Maybe @rpigott or @poettering can seek to resolve this properly later.

Reply:
@bluca Please cut out those personal attacks. This projects strives for a higher standard.

Reply:
As promised I've just opened https://github.com/systemd/systemd/issues/33815 where the part of https://github.com/systemd/systemd/issues/32990 that hasn't been addressed is described. There are other bugs where it violates the RFC and things like that but they aren't related to that PR.

> how it was brought up

@YHNdnzj let me just say I'm not happy about the attitude towards quality coming from some places. As far as I understand there is an opportunity to improve the resolved testing infrastructure but instead of, say, integrating the bonjour conformance test that can exercise the mDNS part of resolved (which is totally untested) the effort is focused on bumping the coverage. It isn't going to help with any of the issues mentioned here. It kind of makes sense but it would make much more sense if stuff was run under Valgrind for example. Either way I don't know what the plan is there and nobody replies to my questions as usual.